### PR TITLE
feat(AIP-148): use uid instead of id

### LINKED ISSUE #1068: AIP-148: Recommend uid instead of id
In http://aip.dev/148, the field `uid` is considered a standard field (used by http://aip.dev.122). When there is an `string id` field on a resource message, it should be suggested as `uid` instead.

### PR DESCRIPTION
**Pull request by noahdietz**
The standard field for the Resource ID segment within a resource is `uid`, not `id` per AIP-148. So let's enforce that.

Fixes #1068

### CODE REVIEW

File: docs/rules/0148/use-uid.md [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+---
+
+# Human names
+
+This rule encourages the use of `uid` instead of `id` for resource messages, as
+mandated in [AIP-148][].
+
+## Details
+
+This rule looks for fields named `first_name` and `last_name`, and complains if
```
Comment (toumorokoshi) [inline docs/rules/0148/use-uid.md:18]: looks like this should be changed?

File: docs/rules/0148/use-uid.md [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+(respectively) instead.
+
+## Examples
+
+**Incorrect** code for this rule:
+
+```proto
+// Incorrect.
+message Human {
+  string first_name = 1;  // Should be `given_name`.
```
Comment (toumorokoshi) [inline docs/rules/0148/use-uid.md:29]: change examples too?

File: docs/rules/0148/use-uid.md [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+```
+
+## Disabling
+
+If you need to violate this rule, use a leading comment above the field or its
+enclosing message. Remember to also include an [aip.dev/not-precedent][]
+comment explaining why.
+
+```proto
+// (-- api-linter: core::0148::human-names=disabled
```
Comment (toumorokoshi) [inline docs/rules/0148/use-uid.md:51]: change api linter example?

side note: I think we could generate a ton of this documentation with the right metadata.

File: rules/aip0148/use_uid_test.go [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+	"github.com/googleapis/api-linter/rules/internal/testutils"
+)
+
+func TestUseUid(t *testing.T) {
+	for _, test := range []struct {
+		FieldName string
+		problems  testutils.Problems
+	}{
+		{"uid", nil},
+		{"id", testutils.Problems{{Suggestion: "uid"}}},
```
Comment (toumorokoshi) [inline rules/aip0148/use_uid_test.go:29]: maybe a quick test case for an unrelated field (e.g. `foo_id`?)

**Review by toumorokoshi — CHANGES_REQUESTED** [review]
Thanks! I think the example and docs need to be updated, but the code LGTM.

File: rules/aip0148/use_uid_test.go (hunk 4 above)
Comment (noahdietz) [reply rules/aip0148/use_uid_test.go:29]: Done.

File: docs/rules/0148/use-uid.md (hunk 3 above)
Comment (noahdietz) [reply docs/rules/0148/use-uid.md]: Fixed

File: docs/rules/0148/use-uid.md (hunk 2 above)
Comment (noahdietz) [reply docs/rules/0148/use-uid.md]: Done.

File: docs/rules/0148/use-uid.md (hunk 1 above)
Comment (noahdietz) [reply docs/rules/0148/use-uid.md]: Ha yeah, forgot to update this completely. Done.

File: rules/aip0148/use_uid.go [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+)
+
+var useUid = &lint.FieldRule{
+	Name: lint.NewRuleName(148, "use-uid"),
+	OnlyIf: func(f *desc.FieldDescriptor) bool {
+		return utils.IsResource(f.GetOwner())
+	},
+	LintField: func(f *desc.FieldDescriptor) []lint.Problem {
+		want := "uid"
+		target := "id"
```
[the remark this answers is not in the document]
Comment (noahdietz) [reply rules/aip0148/use_uid.go]: Done! Good idea.

**Review by toumorokoshi — APPROVED** [review]
Awesome, thanks!

**Review by shwoodard — APPROVED** [review]

**Pull request merged**